packages/client/src/internal/client/methods.ts

```diff
@@ -53,14 +52,17 @@ export class ClientMethods extends ClientCore implements IClientMethods, IClient
     public async getMileageBalances(email: string): Promise<BigNumber> {
         if (!checkEmail(email)) throw new InvalidEmailParamError();
 
-        const provider = this.web3.getProvider() as Provider;
+        const provider = this.web3.getProvider();
+        if (!provider) throw new NoProviderError();
```

Comment:
const signer = this.web3.getConnectedSigner();
        if (!signer) {
            throw new NoSignerError();
        } else if (!signer.provider) {
            throw new NoProviderError();
        }